### PR TITLE
Use tox to run multi-platform tests

### PR DESCRIPTION
**Pull request by alexanderson**
@artsalliancemedia/thunderstorm 

Thoughts?
**+** Since the container has the deps for all 3 versions baked in, there's less overhead having to install them all every time tests are run.
**+** Easier to add more versions (also more idiomatic)
**-** Tests aren't run in parallel (but point 1 means it's still faster)

### CODE REVIEW

File: install-prereqs.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
```
Comment (robyoung) [inline install-prereqs.sh]: This file doesn't look like it's getting copied into the container. How does it work?

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[tox]
```
Comment (robyoung) [inline tox.ini]: This is never copied into the container. Should it be?

Comment (alexanderson) [reply tox.ini]: The tox base container does the following:
```
ONBUILD COPY install-prereqs*.sh requirements*.txt tox.ini /app/
ONBUILD ARG SKIP_TOX=false
ONBUILD RUN bash -c " \
    if [ -f '/app/install-prereqs.sh' ]; then \
        bash /app/install-prereqs.sh; \
    fi && \
    if [ $SKIP_TOX == false ]; then \
        TOXBUILD=true tox; \
    fi"
```

File: install-prereqs.sh (hunk 1 above)
Comment (alexanderson) [reply install-prereqs.sh]: See below about base image.

File: tox.ini (hunk 2 above)
Comment (robyoung) [reply tox.ini]: ahhh, I see :+1:

**Comment by alexanderson** [comment]
> --junitxml must be a filename, given: results.xml
are you kidding me?! I despise Jenkins

File: Makefile [hunk 3]
```diff
@@ -12,9 +12,9 @@ lint:
 
 test: lint
 	pytest \
-		--cov thunderstorm_auth \
-		--cov-report xml \
-		--junit-xml=results.xml \
+		--cov thunderstorm_auth\
+		--cov-report xml\
+		--junit-xml results.xml\
```
Comment (shipperizer) [inline Makefile]: you are missing the `=`, maybe that's the reason why it s failing

**Comment by will-norris** [comment]
:+1:  for this, waaaaay better than the current solution

File: Makefile (hunk 3 above)
Comment (alexanderson) [reply Makefile]: It failed with it too, hence why I tried taking it out

Comment (alexanderson) [reply Makefile]: So the issue is the file doesn't exist on the host, so can't be a volume.
We can mount the whole repo directory into the container, but that also mounts my `.python-verison` file in, confusing `pyenv` in the container.
Could potentially use `docker cp` to extract the file?

Comment (shipperizer) [reply Makefile]: u can mount only the files u care, like *.py

Comment (alexanderson) [reply Makefile]: @shipperizer 
Then we would need an empty results file in the repo to allow the container to write to it. If we attempt to mount a non-existent file it complains.

Comment (alexanderson) [reply Makefile]: But including an empty results file in the repo means removing it from .gitignore, and risks results from local test runs being committed.

**Pull request merged**